ironfish-rust/src/frost_utils/split_secret.rs

```diff
@@ -30,25 +31,35 @@ pub(crate) fn split_secret<R: RngCore + CryptoRng>(
 
     let secret_key = SigningKey::deserialize(secret_bytes)?;
 
-    let identifier_list = IdentifierList::Custom(&config.identifiers);
+    let mut frost_id_map = config
+        .identifiers
+        .iter()
+        .cloned()
+        .map(|identity| (identity.to_frost_identifier(), identity))
+        .collect::<HashMap<_, _>>();
+    let frost_ids = frost_id_map.keys().cloned().collect::<Vec<_>>();
+    let identifier_list = IdentifierList::Custom(&frost_ids[..]);
 
     let (shares, pubkeys) = split(
         &secret_key,
-        config.identifiers.len() as u16,
+        config
+            .identifiers
+            .len()
+            .try_into()
+            .expect("too many identifiers"),
         config.min_signers,
         identifier_list,
         &mut rng,
     )?;
 
-    for (_k, v) in shares.clone() {
-        KeyPackage::try_from(v)?;
-    }
-
     let mut key_packages: HashMap<_, _> = HashMap::new();
 
-    for (identifier, secret_share) in shares {
+    for (frost_id, secret_share) in shares {
+        let identity = frost_id_map
+            .remove(&frost_id)
+            .expect("frost returned an identifier that was not passed as an input");
```

Comment:
Should we return an `IronfishError` instead of expect?

Reply:
This would be a symptom of a bug in the `frost` crate, not something that depends on user input, so I'd say that a panic here is fine. The rule I'm following is "panic for internal bugs, results for input and system errors".